File: 2379-minimum-recolors-to-get-k-consecutive-black-blocks/2379-minimum-recolors-to-get-k-consecutive-black-blocks.java
class Solution {
    public int minimumRecolors(String blocks, int k) {
        int n = blocks.length();
        int minop = n;
        for(int i=0;i<=n-k;i++){
            int white = 0;
            for(int j = i;j<i+k;j++){
                if(blocks.charAt(j)=='W'){
                    white++;
                }
            }
            minop = Math.min(white,minop);
        }
        return minop;
        
    }
}

File: 2379-minimum-recolors-to-get-k-consecutive-black-blocks/2379-minimum-recolors-to-get-k-consecutive-black-blocks.cpp
class Solution {
public:
    int minimumRecolors(string blo, int k) {
       int ans = 0;
        int wc = 0;
        for(int i=0;i<k;i++){
            if(blo[i] == 'W'){
                wc++;
            }   
        }
        ans = wc;
        for(int i=k;i<blo.size();i++){
            if(blo[i] == 'W'){
                wc++;
            }
            if(blo[i-k] == 'W'){
                wc--;
            }
            ans = min(ans,wc);
        }
        return ans;
    }
};